subsys/net/lib/nrf_cloud/include/nrf_cloud_transport.h

```diff
@@ -90,6 +90,16 @@ int nct_dc_send(const struct nct_dc_data *dc);
  */
 int nct_dc_stream(const struct nct_dc_data *dc);
 
+/** @brief Publish data on the batch endpoint topic.
+ *
+ *  @param[in] dc_data Pointer to structure containing the data to be published.
+ *  @param[in] qos MQTT Quality of Service level of the publication.
+ *
+ *  @return 0 If successful. Otherwise, a negative error code is returned.
```

Comment:
```suggestion
 *  @retval 0 If successful. Otherwise, a negative error code is returned.
```

Not sure if combination of return and retval is intentional.

Reply:
It is intentional. `0 If successful. Otherwise, a negative error code is returned` is more of a generic return description rather than a pure `retval` description. I usually add a few `retvals` for the more common error related return values.

Reply:
Out of curiosity, do you have some example on how does it render in the final documenation? I always thought one should not mix `return` and `retval` tags.

Reply:
`aws_fota_init()` for example, documented here; http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-5676/nrf/libraries/networking/aws_fota.html#c.aws_fota_init.

_I always thought one should not mix return and retval tags._

I wasn't aware of that. :(

Reply:
Thanks. I see that we end up with two "Returns:" sections for the function (one from `return` tag and other from `retval`s). I'm not sure if that's desired though, I'll leave it for tech writers to judge on that.

Reply:
@b-gent Could you suggest?

Reply:
We don't have that defined in the styleguide but I believe having both ``@return`` and ``@retval`` for one function/structure is ok. Return is used for more descriptive documentation. ``@retval`` for specifically providing return values and documenting when they occur.
Usually, when ``@retval`` is used, there should be more than one return value documented. So in this case, I would agree with @simensrostad that we can treat it as a generic description and keep as ``@return``.

I was not aware that output of both ``@return`` and ``@retval`` is the same and shows up as "Returns"
![image](https://user-images.githubusercontent.com/29537761/134867009-3b87134d-0b07-4971-87eb-89b2138917b4.png)

I don't think this is good and will create a Jira task to update the doxygen stylesheet. Rendered ``@retval`` should have a "Return values" heading, like it used to in nRF5 SDK
![image](https://user-images.githubusercontent.com/29537761/134867545-b83a8481-7342-435f-9550-993fe3c3d218.png)